src/components/Filters/index.tsx

```diff
@@ -0,0 +1,245 @@
+	onFiltersChange?: (filters: JSONSchema[]) => void;
+	onViewsChange?: (views: FiltersView[]) => void;
+	renderMode?: FilterRenderMode | FilterRenderMode[];
+	onSearch?: (searchTerm: string) => React.ReactElement | null;
+	showSaveView?: boolean;
+}
+
+export const Filters = ({
+	schema,
+	dark,
+	filters,
+	views,
+	onFiltersChange,
+	onViewsChange,
+	renderMode,
+	onSearch,
+	showSaveView,
+}: FiltersProps) => {
+	const { t } = useTranslation();
+	const theme = useTheme();
+	const matches = useMediaQuery(theme.breakpoints.up('sm'));
+	const [showFilterDialog, setShowFilterDialog] = React.useState(false);
+	const [showSearchDropDown, setShowSearchDropDown] = React.useState(false);
+	const [searchString, setSearchString] = React.useState<string>('');
+	const [editFilter, setEditFilter] = React.useState<JSONSchema>();
+	const [formData, setFormData] = React.useState<FormData[] | FormData>();
+	const searchBarRef = useClickOutside<HTMLDivElement>(() =>
+		setShowSearchDropDown(false),
+	);
+
+	React.useEffect(() => {
+		if (!editFilter) {
+			return;
+		}
+		if (editFilter.title === FULL_TEXT_SLUG) {
+			const description = parseFilterDescription(editFilter);
+			if (!description) {
+				return;
+			}
+			setFormData({
+				field: description.field,
+				operator: 'contains',
+				value: description.value,
+			});
+			return;
+		}
+		setFormData(
+			editFilter.anyOf
+				?.map((f) => {
+					if (!isJSONSchema(f)) {
+						return;
+					}
+					const description = parseFilterDescription(f);
+					if (!description) {
+						return;
+					}
+					return {
+						field: description.field,
+						operator: description.operator,
+						value: description.value,
+					};
+				})
+				.filter((f) => !!f) as FormData[],
+		);
+	}, [editFilter, setFormData]);
+
+	const handleFilterChange = React.useCallback(
+		(filter: JSONSchema | null | undefined) => {
+			if (!filter) {
+				return;
+			}
+			let existingFilters = filters;
+			if (editFilter && filters) {
+				existingFilters = filters.filter((f) => f.$id !== editFilter.$id);
+			}
+			onFiltersChange?.([...(existingFilters ?? []), filter]);
+		},
+		[filters, onFiltersChange, editFilter],
+	);
+
+	const handleDeleteFilter = React.useCallback(
+		(filter: JSONSchema) => {
+			const newFilters = filters?.filter((f) => f.$id !== filter.$id) ?? [];
+			onFiltersChange?.(newFilters);
+		},
+		[filters, onFiltersChange],
+	);
+
+	const handleSaveView = React.useCallback(
+		(name: string) => {
+			const view: FiltersView = {
+				id: randomUUID(),
+				name,
+				eventName: `Saving filters view ${name}`,
+				filters: filters ?? [],
+			};
+			const newViews = [...(views ?? []), view];
+			onViewsChange?.(newViews);
+		},
+		[filters, views, onViewsChange],
+	);
+
+	const handleDeleteView = React.useCallback(
+		(view: FiltersView) => {
+			const newViews = views?.filter((v) => v.id !== view.id);
+			onViewsChange?.(newViews ?? []);
+		},
+		[views, onViewsChange],
+	);
+
+	return (
+		<Box>
```

Comment:
nit: Just curious why we went with a Box instead of fragment?

Reply:
no real reasons, just have a wrapper but we can for sure use a fragment, let me replace it 👍